pkg/burner/job.go

```diff
@@ -242,9 +250,16 @@ func Run(configSpec config.Spec, kubeClientProvider *config.KubeClientProvider,
 	}()
 	select {
 	case rc = <-res:
-	case <-time.After(timeout):
-		err := fmt.Errorf("%v timeout reached", timeout)
+	case <-time.After(configSpec.GlobalConfig.Timeout):
+		err := fmt.Errorf("%v timeout reached", configSpec.GlobalConfig.Timeout)
 		log.Error(err.Error())
+		executedJobs[len(executedJobs)-1].End = time.Now().UTC()
+		gcCtx, cancel := context.WithTimeout(context.Background(), globalConfig.GCTimeout)
+		defer cancel()
+		for _, job := range jobList {
```

Comment:
Here I think we need to only iterate through the list of jobs that have executed. Because in the entire list of jobs (.i.e for example in total 4 jobs) only 1 got executed and the 2nd got timed out, we need to do GC only till those first 2 jobs that actually deployed resources on the cluster. 3rd and 4th are not required.
```suggestion
		for _, job := range jobList[len(executedJobs)-1] {
```